content/blog/2021/containerqueries.md

```diff
@@ -0,0 +1,146 @@
+
+ Bookmark Miriam's scratch site for updates: [Miriam's CSS Sandbox](https://css.oddbird.net/rwd/query/). 
+
+ ## Background
+
+`@Media` queries sparked a responsive design revolution by allowing authors to change the styles of elements based on the size of the entire viewport. Up until now, what could not be done was changing the style of an element based on the size of its parent element. 
+
+## Syntax and an example
+
+ The proposed `@container` query syntax feels a lot like writing a `@media` query, but a main difference is that a `@container` query has to be implemented in two parts.
+
+In this demo (which only works in the latest version of [Chrome Canary](https://www.google.com/chrome/canary)), a `blockquote` is styled differently depending on the size of its container. 
+
+{{ embed.codepen(
+  id='YzpywrZ',
+  title='Container Queries with Block Quotes',
+  user='dvdherron',
+  height=600
+) }}
+
+### Define a containment context
+
+To achieve this, first a containment context must be defined. This lets the browser know which container to query against later. 
+
+```scss
+article,
+section {
+  contain: layout inline-size;
+}
+```
+
+The `contain` property is part of the existing [CSS Containment Module](https://drafts.csswg.org/css-contain/). The `layout` value turns on [layout contaiment](https://drafts.csswg.org/css-contain/#valdef-contain-layout) on the container, which ensures that "nothing outside can affect its internal layout, and vice versa." The `inline-size` value is a proposed change to the Containment Module that would let authors explicitly declare in which dimension containment should be applied. 
+
+### Apply @-rules at desired breakpoints
+Now that the containment context has been defined, the `@container` rule is used to tell the browser when and how styles should change. 
+
+```scss
+/* change styles according to container size */
+@container (min-width: 35em) {
+
+  .first {
+    --column: full;
+    margin-bottom: 0.5rem;
+  }
+
+  .title {
+    --title-background: forestgreen;
+  }
+
+  [data-quote] {
+    grid-gap: 1rem;
+    grid-template: "media quote" auto/ max-content 1fr;
+  }
+
+  .media {
+    align-items: center;
+  }
+
+  .source {
+    font-weight: bold;
+  }
+}
+
+@container (min-width: 40em) {
+
+  .spread {
+    --column: full;
+    display: grid;
+    grid-template-columns: minmax(50%, 1fr) minmax(min-content, 20em);
+    gap: 2rem;
+  }
+}
+
+@container (min-width: 60em) {
+
+  .title {
+    --title-background: goldenrod;
+  }
+
+  [data-quote] {
+    --quote-size: 1.5rem;
+    --quote-background: darkmagenta;
+    clip-path: polygon(
+      -0.04% 100.6%,
+      1.12% -0.52%,
+      95.65% 11.18%,
+      96.46% 90.05%
+    );
+    padding: 1.25em 3em;
+    max-width: unset;
+  }
+
+  .source {
+    font-size: 1.1rem;
+  }
+}
+```
```

Comment:
This is a long block of code. Consider either simplifying to particular examples, or breaking it apart into several blocks, and walking the reader through the changes?

Reply:
Good call. Broke this up.